SeQuant/core/type_traits.hpp

```diff
@@ -0,0 +1,47 @@
+//
```

Comment:
meta programming (including type traits) are already present in `meta.hpp`, should go there, unless there is a good reason. not to. Also, to be "safe" meta programming lives in `sequent::meta` namespace since most of the time users do not need to bother